### ISSUE TITLE
ValueError : 1 is not a valid Actions

### ISSUE THREAD

**Issue by HARIHARAN1103** [LZQthePlane/Online-Realtime-Action-Recognition-based-on-OpenPose#9]
I have trained the model using only one action.Training for one action is successfully done.While testing , the webcam is running for about 2 seconds and it is getting terminated.And I got this error.
![Screenshot (232)](https://user-images.githubusercontent.com/35688511/54176536-417d2280-44b5-11e9-9131-fbe2a0600dd1.png)

**Comment by LZQthePlane** [comment]
I don't know if you change the `Action` enum class right. 
If you did it correctly, then I have no idea, either.

**Comment by HARIHARAN1103** [comment]
In action_enum.py I have kept only one class. I have attached the .py files where I have made some changes.Please check if its right.

**My train.py file:**
`import pandas as pd
from enum import Enum
import numpy as np
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split
from keras.utils import np_utils
from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras.layers.normalization import BatchNormalization
from keras.optimizers import Adam
from keras.models import load_model


class Actions(Enum):
    # framewise_recognition.h5
    # squat = 0
    # stand = 1
    # walk = 2
    # wave = 3

    # framewise_recognition_under_scene.h5
    sit = 0
    #walk = 1
    #sleep= 2
    #fall_down = 3
	#run = 4


# load data
raw_data = pd.read_csv('sit.csv', header=0)
dataset = raw_data.values
# X = dataset[:, 0:36].astype(float)
# Y = dataset[:, 36]
X = dataset[0:815, 0:36].astype(float)  # 忽略run数据
Y = dataset[0:815, 36]


#encoder_Y = [0]*744 + [1]*722 + [2]*815 + [3]*1008 + [4]*811
#encoder_Y = [0]*0 + [1]*0 + [2]*815 + [3]*0 + [4]*0
encoder_Y = [0]*815
# 将类别编码为数字
#encoder = LabelEncoder()
#encoder_Y = encoder.fit_transform(Y)
# print(encoder_Y[0], encoder_Y[900], encoder_Y[1800], encoder_Y[2700])
# one hot 编码
dummy_Y = np_utils.to_categorical(encoder_Y)


# train test split
X_train, X_test, Y_train, Y_test = train_test_split(X, dummy_Y, test_size=0.1, random_state=9)

# build keras model
model = Sequential()
model.add(Dense(units=128, activation='relu'))
model.add(BatchNormalization())
model.add(Dense(units=64, activation='relu'))
model.add(BatchNormalization())
model.add(Dense(units=16, activation='relu'))
model.add(BatchNormalization())
model.add(Dense(units=1, activation='softmax'))

# training
model.compile(loss='sparse_categorical_crossentropy', optimizer=Adam(0.0001), metrics=['accuracy'])
model.fit(X_train, Y_train, batch_size=32, epochs=50, verbose=1, validation_data=(X_test, Y_test))
model.save('framewise_recognition.h5')
`
**My action_enum.py file:**

`from enum import Enum


class Actions(Enum):
    """
    Actions enum
    """
    # framewise_recognition.h5
    # squat = 0
    # stand = 1
    # walk = 2
    # wave = 3

    # framewise_recognition_under_scene.h5
    sit = 0
    #walk = 1
    #sleep = 2
    #fall_down = 3
    #run = 4
`

**Comment by LZQthePlane** [comment]
I am not sure, may be because of the one hot encoding in line `dummy_Y = np_utils.to_categorical(encoder_Y)`, 
just try to debug step by step, Good luck to u

**Comment by zhangdengp** [comment]
i also get the same problem. 
'2 is not a valid Actions'

**Comment by don0979** [comment]
I have the same problem.
'2 is not a valid Actions'
Would you like to give me some suggestion? Thank you!
